[Iss306] update load wra data model

BiancaMorandi opened the pull request:
The updates below were made:

1. make code work with latest version of the WRA data model (test performed also with the current version in dev) https://github.com/IEA-Task-43/digital_wra_data_standard/blob/dev/schema/iea43_wra_data_model.schema.json

    1.  update field names when needed (e.g logger_main_config, logger_measurement_config, floating_lidar etc)
    2.  fixed bug not recognising definitions when ending with _id

1. updated `_filter_parent_level` to do not remove properties if None
2. updated `measurements.get_table()` to remove columns if all values are None
3. updated demo file for testing WRA data model to version v1.0
4. added tests in `test_station.py`

stephenholleran commented:
@BiancaMorandi anything to do here with regards the new version 1.2? Testing or anything?

stephenholleran commented on brightwind/load/station.py:
```diff
@@ -115,15 +115,18 @@ def _rename_to_title(list_or_dict, schema):
             prefixed_name = key + PREFIX_DICT[key]['prefix_separator'] + col
             prefixed_names[prefixed_name] = {'prefix_separator': PREFIX_DICT[key]['prefix_separator'],
                                              'title_prefix': PREFIX_DICT[key]['title_prefix']}
+    list_special_cases_no_prefix = ['sensor_config.slope', 'sensor_config.offset', 'sensor_config.sensitivity',
```
Do we need to keep the 'sensor_config' ones? Just support from v1.0 onwards.

BiancaMorandi replied:
solved

stephenholleran commented on brightwind/load/station.py:
```diff
@@ -307,6 +309,13 @@ def _raise_child(dictionary, child_to_raise):
+    'logger_measurement_config': {
+        'prefix_separator': '.',
+        'title_prefix': 'Logger ',
+        'keys_to_prefix': ['height_m', 'height_reference_id', 'serial_number',
```
Is 'height_reference' part of logger measurement config?

BiancaMorandi replied:
solved

stephenholleran commented on brightwind/load/station.py:
```diff
@@ -307,6 +309,13 @@ def _raise_child(dictionary, child_to_raise):
+        'title_prefix': 'Logger ',
+        'keys_to_prefix': ['height_m', 'height_reference_id', 'serial_number',
+                           'slope', 'offset', 'sensitivity',
+                           'notes', 'update_at']
```
Not sure why having 'notes' and 'update_at' adds here?

stephenholleran commented on brightwind/load/station.py:
```diff
@@ -757,25 +766,25 @@ def wdir_heights(self):
-        :return:              The properties merged together.
-        :rtype:               list(dict)
+        :param logger_measurement_cfgs: Sensor cfgs properties
+        :type logger_measurement_cfgs:  list
```
We should probably use 'configs' instead of 'cfgs'.

BiancaMorandi replied:
renamed all to logger_measurement_configs

stephenholleran commented on brightwind/load/station.py:
```diff
@@ -814,9 +824,9 @@ def __meas_point_merge(sensor_cfgs, sensors=None, mount_arrgmts=None):
-            # col_names_raised = _raise_child(meas_point, child_to_raise='column_name')
-            # sen_cfgs = _raise_child(col_names_raised, child_to_raise='sensor_config')
-            sen_cfgs = _raise_child(meas_point, child_to_raise='sensor_config')
+            logger_meas_cfgs = _raise_child(meas_point, child_to_raise='sensor_config')
```
How does 'sensor_config' work in this line? There shouldn't be any 'sensor_config'.

BiancaMorandi replied:
removed

BiancaMorandi commented:
@stephenholleran I have addressed all your comments:

1. removed `sensor_config`
2. renamed `logger_measurement_config(s)`
3. renamed `mounting arrangements`
